hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -1040,6 +1035,32 @@ private static Object rewritePrimaryTypeWithDiffSchemaType(Object oldValue, Sche
     throw new AvroRuntimeException(String.format("cannot support rewrite value for schema type: %s since the old schema type is: %s", newSchema, oldSchema));
   }
 
+  /**
+   * When there is a lost in scale, rounding is required to be performed when performing casts.
+   * For types that have a fix scale, the results of the cast will need to be the same for COW and MOR.
+   * <p>
+   * NOTE: COW uses Spark's default rounding of HALF_UP (if Spark entry points are used).
+   * <p>
+   * Floating point types are not considered here as they are susceptible to floating point rounding errors.
+   * <p>
+   * Summary:
+   * <ul>
+   *   <li>double/string -> decimal: HALF_UP</li>
+   *   <li>float -> decimal: HALF_EVEN</li>
+   * </ul>
```

Comment:
spark always use HALF_UP to cast  FractionalType（float/double） to DecimalType
why we use HAL_EVEN to cast float -> decimal     
BTW  hive/presto also use HALF_UP

Reply:
@xiarixiaoyao The original code that i am modifying uses `HALF_EVEN`. 

According to the git-blame, the `HALF_EVEN` rounding mode was introduced by @alexeykudinkin.

Reply:
I think we should unify all the rounding mode to `HALF_UP` to avoid ambiguity. And another confusion is #7769, all the numeric and string types use the `HALP_EVEN`, why this patch only fixes `DOUBLE` and `STRING` type, the strategy are out of sync.

Reply:
My initial considerations for only applying this to `DOUBLE` and `STRING` was because these are exact representation of fractional numbers.
i.e. 1.5 = 1.5.

For any other types, like float, these are not exact. 
i.e. 1.5 is usually represented as 1.49918237

As such, given that `HALF_EVEN` was introduced in #7769, i wanted to limit the change to the types to exact types.

Since we want to standardize all rounding mode to `HALF_UP`, this method is no long required then. I will standardise everything to `HALF_UP`.